### PR TITLE
Add z-changelog; replacement for scripts/Get-Changelog.ps1

### PR DESCRIPTION
**Pull request by BillyONeal**
* Built on the 'bones' of `portsdiff`
* Also implemented the single pass algorithm for `portsdiff`
* Fixed bad newlines in `portsdiff`:

example output

* Removed the version numbers for the 'removed' ports output because the version at the previous snapshot is meaningless; the meaningful version would be the last version but we don't know that.

### CODE REVIEW

File: docs/vcpkg_registry_release_process.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# vcpkg_registry_release_process
+
+This document describes the acceptance criteria / process we use when doing a vcpkg registry release.
+
+1. Run vcpkg z-changelog `<SHA OF LAST RELEASE>` > `path/to/results.md`
+1. Create a new GitHub release in the registry repo.
+1. Run 'auto generate release notes'
+1. Change `## New Contributors` to `#### New Contributors`
+1. Copy the contents to the end of `path/to/results.md` (the `#### New Contributors` part should line up)
+1. Change the link to the most recent CI build, and copy number of successful ports into the table at the top.
```
Comment (ras0219-msft) [inline docs/vcpkg_registry_release_process.md:10]: This seems likely to cause a mistake (forget to fill in the correct final table); I'd drop it.

File: src/vcpkg/commands.z-changelog.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        if (!portsdiff.added_ports.empty())
+        {
+            result.append("<details>\n");
+            result.append(fmt::format("<summary><b>The following {} ports have been added:</b></summary>\n\n",
+                                      portsdiff.added_ports.size()));
+            result.append("|port|version|\n");
+            result.append("|---|---|\n");
+            for (auto&& added_port : portsdiff.added_ports)
+            {
+                result.append(fmt::format("|{}|{}|\n", added_port.port_name, added_port.version));
```
Comment (ras0219-msft) [inline src/vcpkg/commands.z-changelog.cpp:70]: Is there a reasonable way to use `fmt::format_to()` and a back_inserter here? Something like
```suggestion
                // out_it defined as `auto out_it = std::back_inserter(result);`
                fmt::format_to(out_it, "|{}|{}|\n", added_port.port_name, added_port.version);
```

**Review by ras0219-msft — APPROVED** [review]

File: docs/vcpkg_registry_release_process.md (hunk 1 above)
Comment (BillyONeal) [reply docs/vcpkg_registry_release_process.md:10]: Are you saying to drop the table entirely? I think @AugP said he liked the table...

File: src/vcpkg/commands.z-changelog.cpp (hunk 2 above)
Comment (BillyONeal) [reply src/vcpkg/commands.z-changelog.cpp:70]: Yes, but perf super doesn't matter and I was lazy. Stand by...

Comment (BillyONeal) [reply src/vcpkg/commands.z-changelog.cpp]: Done!

**Review by data-queue — COMMENTED** [review]
May be out of scope for this PR. Are there tests for `portsdiff`? I feel like we need to add them if not: `portsdiff` is not an experimental command, which means it should be supported on all platforms. I was not able to get `portsdiff` working on Linux locally.

**Comment by BillyONeal** [comment]
> May be out of scope for this PR. Are there tests for `portsdiff`? I feel like we need to add them if not: `portsdiff` is not an experimental command, which means it should be supported on all platforms. I was not able to get `portsdiff` working on Linux locally.

I agree tests would be good but I don't have great ideas on doing so :/

**Comment by BillyONeal** [comment]
> May be out of scope for this PR. Are there tests for `portsdiff`? I feel like we need to add them if not: `portsdiff` is not an experimental command, which means it should be supported on all platforms. I was not able to get `portsdiff` working on Linux locally.

```
[DEBUG] 1001: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg -c core.autocrlf=false cat-file -t HEAD)
[DEBUG] 1001: cmd_execute_and_stream_data() returned 0 after     1479 us
[DEBUG] 1002: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg -c core.autocrlf=false cat-file -t HEAD~10)
[DEBUG] 1002: cmd_execute_and_stream_data() returned 0 after     1371 us
[DEBUG] 1003: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg/ports-HEAD -c core.autocrlf=false checkout HEAD -f -q -- ".\\ports" .vcpkg-root)
[DEBUG] 1003: cmd_execute_and_stream_data() returned 1 after     4097 us
[DEBUG] 1004: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg/ports-HEAD -c core.autocrlf=false reset)
[DEBUG] 1004: cmd_execute_and_stream_data() returned 0 after    14753 us
[DEBUG] 1005: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg/ports-HEAD~10 -c core.autocrlf=false checkout HEAD~10 -f -q -- ".\\ports" .vcpkg-root)
[DEBUG] 1005: cmd_execute_and_stream_data() returned 1 after     4810 us
[DEBUG] 1006: execute_process(/usr/bin/git --git-dir=/home/bion/vcpkg/.git --work-tree=/home/bion/vcpkg/ports-HEAD~10 -c core.autocrlf=false reset)
[DEBUG] 1006: cmd_execute_and_stream_data() returned 0 after    14850 us
```

Looks like there's a `\\` that needs to be `/` here

**Review by data-queue — APPROVED** [review]

**Pull request merged**